### PR TITLE
[20791] sustainml_cpp architecture update v2 

### PR DESCRIPTION
**Pull request by JesusPoderoso**
This PR updates the SustainML node library based on the latest design update.
It also fixes the types regeneration script, and adds some instructions to do that task.

**IMPORTANT**: tests have not been fixed, so most of them will fail.

### CODE REVIEW

File: sustainml_cpp/examples/CliModules/main.cpp [hunk 1]
```diff
@@ -172,21 +194,23 @@ class UserInputPublisher {
...
+            int task_id)
     {
         UserInputImpl ui;
         GeoLocationImpl geo;
         geo.continent("Continent of task id " + std::to_string(task_id));
         geo.region("Region of task id " + std::to_string(task_id));
         ui.geo_location(geo);
-        ui.problem_description("Problem definition of task id " + std::to_string(task_id));
+        ui.problem_definition("Problem definition of task id " + std::to_string(task_id));
         ui.task_id(task_id);
```
Comment (Mario-DL) [inline sustainml_cpp/examples/CliModules/main.cpp:206]: Here we may want to fill a task_name including the `to_string(task_id)` within the name, to make it change across tasks

File: sustainml_cpp/include/sustainml_cpp/nodes/MLModelMetadataNode.hpp [hunk 2]
```diff
@@ -36,35 +36,35 @@ namespace utils {
 template<class T> class SamplePool;
 } // namespace utils
 
-namespace ml_task_encoding_module {
+namespace ml_model_metadata_module {
```
Comment (Mario-DL) [inline sustainml_cpp/include/sustainml_cpp/nodes/MLModelMetadataNode.hpp:39]: We may consider using the same namespace as for the MLModelNode, the `::ml_model_provider` as both of them are related

File: sustainml_cpp/include/sustainml_cpp/nodes/HardwareConstraintsNode.hpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+namespace sustainml {
+namespace core {
+template<class T> class QueuedNodeListener;
+} // namespace core
+
+namespace utils {
+template<class T> class SamplePool;
+} // namespace utils
+
+namespace hw_constraints_module {
```
Comment (Mario-DL) [inline sustainml_cpp/include/sustainml_cpp/nodes/HardwareConstraintsNode.hpp:39]: We may think of using the same namespace as for the hwResourcesNode, the `hw_module`

**Review by Mario-DL — CHANGES_REQUESTED** [review]
Nice upgrade refactor.
Apart from the already mentioned changes:
* Pass the name in the `prepare_new_entry` 
* Resolve some states issue with the new nodes
* Work on an intermediate branch

I am leaving some minor suggestions. Also, I am missing the change on the `on_new_node_output` OrchestratorNodeHandle API to incorporate the new `TaskId` argument (it would pass 3: TaskId, NodeId, data, see design). 

In the next PR I will fix the failing tests and will also try to at least prepare the infrastructure to handle multiple iterations, adding the `/baseline` topics (in another third PR)

**Comment by Mario-DL** [comment]
> Nice upgrade refactor. Apart from the already mentioned changes:
> 
> * Pass the name in the `prepare_new_entry`
> * Resolve some states issue with the new nodes
> * Work on an intermediate branch
> 
> I am leaving some minor suggestions. Also, I am missing the change on the `on_new_node_output` OrchestratorNodeHandle API to incorporate the new `TaskId` argument (it would pass 3: TaskId, NodeId, data, see design).
> 
> In the next PR I will fix the failing tests and will also try to at least prepare the infrastructure to handle multiple iterations, adding the `/baseline` topics (in another third PR)

My previous comment on the `on_new_node_output` was discarded as the TaskId information can be found within the data message itself as it now travels with it

**Comment by JesusPoderoso** [comment]
Thanks @Mario-DL!!
Commits [c508c4f](https://github.com/eProsima/SustainML/pull/27/commits/c508c4fd2c2123d4435bbd6bcf0df06be80b4cc4), [7de2021](https://github.com/eProsima/SustainML/pull/27/commits/7de20210653cb9c497a683b009996480891fa4a6), [7ff5b29](https://github.com/eProsima/SustainML/pull/27/commits/7ff5b2960fa699d8a92a0310be7d6ba26b784286), [4727d91](https://github.com/eProsima/SustainML/pull/27/commits/4727d9199333fe02f9e5d7848173480af93c6125) LGTM

**Pull request merged**